palette/src/visual.rs

```diff
@@ -0,0 +1,46 @@
+#[cfg(feature = "approx")]
+use approx::AbsDiffEq;
+#[cfg(feature = "approx")]
+use core::borrow::Borrow;
+
+/// Methods to tell the tell about the visual characteristics of the color
+#[cfg(feature = "approx")]
+pub trait VisualColor<T>
```

Comment:
Are these traits still necessary for the Ok* colors? If not, I think we can skip them in order to reduce the size of this PR. I think they deserve a bit more thought and design, and maybe not depend on the `approx` crate, so I would rather postpone them for now.

Reply:
To be completely honest, I'm not sure what to do with them at all. The functionality is useful, but I would like to think more about how this kind of trait should work and how it relates to things like color differences. The approx traits are simple because they are just dumb number comparisons, but something that claims to be "visual" need to back that up a bit. Otherwise it will come back and bite us in the future.

Reply:
I use visually_eq in the tests. I can put them in separate files and make them compile for tests only, if you like. 

They test if two colors are visually equal. Of course implementing [ΔE2000](https://en.wikipedia.org/wiki/Color_difference#CIEDE2000) would be more general and more standard-conformant, but also should be an entirely different task.

Reply:
That would be great, thank you! Let's keep them as test utils.

> Of course implementing [ΔE2000](https://en.wikipedia.org/wiki/Color_difference#CIEDE2000) would be more general and more standard-conformant, but also should be an entirely different task.

Absolutely, there is some difference measurement (it may even be that one) but I would like to expand on it and make it a bit more clear what specification it follows. But that's for another time!